### PR TITLE
Add state migration for removing all values from private domain

### PR DESCRIPTION
**Pull request by turbolent**
⚠️ Depends on #2912 

## Description

After all links got migrated to capability controllers, remove all values from the `private` domain.

- Refactor the existing shared state migration code: Generalize the ability to migrate a path of an account, and make value conversion a particular path migrator that is commonly used
- Introduce a new path migrator which removes all values in the private domain

______

- [ ] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

**Review by dsainati1 — APPROVED** [review]
So we are just deleting every capability that was linked via the private domain? Should we tell developers to prepare for this at all?

**Comment by turbolent** [comment]
@dsainati1 In 1.0, the `private` domain is completely gone, and the concept of a private link is replaced by a capability controller. This migration is just removing the dead capability values which previously all got migrated to capability controllers in #2912

File: migrations/capcons/migration_test.go [hunk 1]
```diff
@@ -1586,3 +1592,111 @@ func TestLinkMigration(t *testing.T) {
...
+		Interface: runtimeInterface,
+	})
+	require.NoError(t, err)
+
+	storeTestPathLinks(t, pathLinks, storage, inter)
+
+	storeTestAccountLinks(accountLinks, storage, inter)
+
+	err = storage.Commit(inter, false)
+	require.NoError(t, err)
```
Comment (SupunS) [inline migrations/capcons/migration_test.go:1662]: Maybe also assert the size of the private domain storage before the migration

**Review by SupunS — APPROVED** [review]

File: migrations/capcons/migration_test.go (hunk 1 above)
Comment (turbolent) [reply migrations/capcons/migration_test.go:1671]: Good idea! Added in 25fdcb41d3c5779b679431d8cb0fc14446ee419e

**Pull request merged**